packages/flutter_tools/lib/src/compile.dart

```diff
@@ -238,6 +239,28 @@ class PackageUriMapper {
   }
 }
 
+List<String> _buildModeOptions(BuildMode mode) {
+  switch (mode) {
+    case BuildMode.debug:
+      return <String>[
+        '-Ddart.vm.product=false',
+        '--bytecode-options=source-positions,local-var-info,debugger-stops,instance-field-initializers,keep-unreachable-code,avoid-closure-call-instructions',
+        '--enable-asserts',
+      ];
+    case BuildMode.profile:
+      return <String>[
+        '-Ddart.vm.product=false',
+        '--bytecode-options=source-positions',
+      ];
+    case BuildMode.release:
+      return <String>[
+        '-Ddart.vm.product=true',
```

Comment:
This is also set depending on 'targetProductVm' at lines 323-324. Consider cleaning up that code.
Also, consider defining '-Ddart.vm.profile=true' depending on the mode.

Reply:
Done.